packages/server-wallet/src/protocols/__test__/defunder.test.ts

```diff
@@ -27,28 +33,29 @@ const testLedgerChannel = TestChannel.create({
 
 let store: Store;
 
-function ensureCloseObjective(
+async function ensureCloseObjective(
   channel: TestChannel,
+  tx: Transaction,
   participantIndex = 0
 ): Promise<DBCloseChannelObjective> {
   // add the closeChannel objective and approve
-  return store.transaction(async tx => {
-    const o = await store.ensureObjective(
-      channel.closeChannelObjective([participantIndex, 1 - participantIndex]),
-      tx
-    );
-    await store.approveObjective(o.objectiveId, tx);
-    return o as DBCloseChannelObjective;
-  });
+  const o = await ObjectiveModel.insert(
+    channel.closeChannelObjective([participantIndex, 1 - participantIndex]),
+    true,
+    tx
+  );
+  await store.approveObjective(o.objectiveId, tx);
```

Comment:
nit: do we need to approve the objective if we're using the `preApproved` flag?

Reply:
good catch, we do not 89a67c7